File: service/src/main/java/org/vsegda/util/TimePeriodUnit.java
package org.vsegda.util;

/**
 * @author Roman Elizarov
 */
public enum TimePeriodUnit {
    SECOND('s', TimeUtil.SECOND),
    MINUTE('m', TimeUtil.MINUTE),
    HOUR('h', TimeUtil.HOUR),
    DAY('d', TimeUtil.DAY),
    WEEK('w', TimeUtil.WEEK);

    private final char code;
    private final long period;

    TimePeriodUnit(char code, long period) {
        this.code = code;
        this.period = period;
    }

    public char code() {
        return code;
    }

    public long period() {
        return period;
    }

  @Override
    public String toString() {
        return "1" + code;
    }
}

File: service/src/main/java/org/vsegda/util/TimePeriodUnit.kt
package org.vsegda.util

enum class TimePeriodUnit(
    @get:JvmName("code") val code: Char,
    @get:JvmName("period") val period: Long
) {
    SECOND('s', org.vsegda.util.SECOND),
    MINUTE('m', org.vsegda.util.MINUTE),
    HOUR('h', org.vsegda.util.HOUR),
    DAY('d', org.vsegda.util.DAY),
    WEEK('w', org.vsegda.util.WEEK);

    override fun toString(): String = "1" + code
}
